docker-compose-rule-junit4/src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -159,8 +158,11 @@ public void before() throws IOException, InterruptedException {
     @Override
     public void after() {
         try {
-            shutdownStrategy().shutdown(this.dockerCompose(), this.docker());
-            logCollector().stopCollecting();
+            shutdownStrategy().stop(this.dockerCompose());
+
+            logCollector().collectLogs(this.dockerCompose());
```

Comment:
Does this mean it just collects logs once at shutdown? As in it no longer streams logs?

Reply:
Correct, this is an explicit tradeoff. See:
- https://github.com/palantir/docker-compose-rule/pull/245#discussion_r215345662
- https://github.com/palantir/docker-compose-rule/pull/245#issuecomment-424485444

Reply:
Ah, I didn't see the previous PR.

I guess if the problem is the `docker-compose logs` process is dying, you could make an `InputStream` that transparently waits until the container is started again, then run `docker-compose -f --tail 0` to get the rest of the logs, maybe using `--timestamps` and some other number of rows to `--tail` to ensure you get them all. What do you think?

Then again, that seems quite fickle and it is true that I for example pretty much always just run `dc logs -f` locally.

Reply:
That seems incredibly complicated. Dealing with restarting containers is very difficult and we're not really giving up anything by collecting logs at the end. I would strongly prefer to keep this implementation simple and have something that work for the 99.9% of normal cases.

Reply:
Fair enough